### ISSUE TITLE
ZMK config question

### ISSUE THREAD

**Issue by benfrain** [codethread/qmk.nvim#21]
When I run :QMKFormat on the file I get `[E01] No keymaps found` so looks like my config is not correct?

Using Lazy and following the docs I have this:

```lua
  {
    "codethread/qmk.nvim",
    config = function()
      ---@type qmk.UserConfig
      local conf = {
        name = "LAYOUT_pretty",
        layout = {
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x _ _ _ _ _ _ _ _ x x x x x x",
          "x x x x x _ _ _ _ _ _ _ _ _ _ x x x x x",
          "_ _ _ _ _ _ x x _ _ _ _ x x _ _ _ _ _ _",
          "_ _ _ _ _ _ _ x _ _ _ _ x _ _ _ _ _ _ _",
          "_ _ _ _ _ x x x _ _ _ _ x x x _ _ _ _ _",
        }
    }
        require("qmk").setup(conf)
    end,
  },
```

I have the devicetree parser installed, and here is the file I am trying to format: https://github.com/benfrain/Adv360-Pro-ZMK/blob/V2.0/config/adv360.keymap

Could you advise what I am doing wrong?

**Comment by codethread** [comment]
hey 😄 there's a missing `variant` config field, there's an [example in the README]E(https://github.com/codethread/qmk.nvim#examples), but it's probably a bit hidden 😅 . The default behaviour assumes QMK, but you're using ZMK

**tl;dr**

```lua
  {
    "codethread/qmk.nvim",
    config = function()
      ---@type qmk.UserConfig
      local conf = {
        name = "LAYOUT_pretty",
		variant = 'zmk', -- <<
        layout = {
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x x _ _ _ _ _ _ x x x x x x x",
          "x x x x x x _ _ _ _ _ _ _ _ x x x x x x",
          "x x x x x _ _ _ _ _ _ _ _ _ _ x x x x x",
          "_ _ _ _ _ _ x x _ _ _ _ x x _ _ _ _ _ _",
          "_ _ _ _ _ _ _ x _ _ _ _ x _ _ _ _ _ _ _",
          "_ _ _ _ _ x x x _ _ _ _ x x x _ _ _ _ _",
        }
    }
        require("qmk").setup(conf)
    end,
  },
```

**Comment by codethread** [comment]
though i will update the code to print more helpful errors for the next person 😅

**Comment by benfrain** [comment]
@codethread ah, I see. Might be worth adding the default `variant = "qmk"` line in the main example too, just so the copy and pasters (like me :)) have a better idea of what they need to change?
